burn-ndarray/src/element.rs

```diff
@@ -13,66 +15,140 @@ pub(crate) trait ExpElement {
     fn sqrt_elem(self) -> Self;
 }
 
-macro_rules! impl_exp_elem {
-    ($elem:ident) => {
-        impl ExpElement for $elem {
-            fn exp_elem(self) -> Self {
-                $elem::exp(self)
-            }
-            fn log_elem(self) -> Self {
-                $elem::ln(self)
-            }
-            fn log1p_elem(self) -> Self {
-                $elem::ln_1p(self)
-            }
-            fn pow_elem(self, value: f32) -> Self {
-                $elem::powf(self, value.into())
-            }
-            fn sqrt_elem(self) -> Self {
-                $elem::sqrt(self)
-            }
-        }
-    };
-    ($elem:ident, $tmp:ident) => {
-        impl ExpElement for $elem {
-            fn exp_elem(self) -> Self {
-                let tmp = $tmp::exp(self as $tmp);
-                tmp as $elem
-            }
-            fn log_elem(self) -> Self {
-                let tmp = $tmp::ln(self as $tmp);
-                tmp as $elem
-            }
-            fn log1p_elem(self) -> Self {
-                let tmp = $tmp::ln_1p(self as $tmp);
-                tmp as $elem
-            }
-            fn pow_elem(self, value: f32) -> Self {
-                let tmp = $tmp::powf(self as $tmp, value as $tmp);
-                tmp as $elem
-            }
-            fn sqrt_elem(self) -> Self {
-                let tmp = $tmp::sqrt(self as $tmp);
-                tmp as $elem
-            }
-        }
-    };
-}
-
 impl NdArrayElement for f64 {}
-impl_exp_elem!(f64);
+impl ExpElement for f64 {
```

Comment:
I prefer having a macro for that kind of stuff, but it can be refactored later.

Reply:
I thought keeping the macros but I would have introduced additional two (four would have been total) to generate 6 implementation. The new would have been needed for f32 which is supported as separate function in libm. Overall I think it was simpler to see code without macros.